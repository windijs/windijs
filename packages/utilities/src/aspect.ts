import { aspectRatioConfig, spacingConfig } from "@windijs/config";
import { configHandler, createUtility, cssHandler, genericHandler, numberHandler } from "@windijs/core";
import { css } from "@windijs/helpers";

import type { CSSObject, StyleObject } from "@windijs/helpers";

const aspectBase: CSSObject = {
  position: "relative",
  "> *": {
    position: "absolute",
    height: "100%",
    width: "100%",
    top: "0",
    right: "0",
    bottom: "0",
    left: "0",
  },
};

export default createUtility("aspect")
  .use(configHandler(aspectRatioConfig, "aspectRatio"))
  .case(
    "none",
    cssHandler({
      position: "static",
      paddingBottom: "0",
      "> *": {
        position: "static",
        height: "auto",
        width: "auto",
        top: "auto",
        right: "auto",
        bottom: "auto",
        left: "auto",
      },
    })
  )
  .case(
    "w",
    numberHandler<Record<keyof typeof spacingConfig, StyleObject> & Record<number, StyleObject>>(v =>
      css({
        "--w-aspect-w": v as string,
        paddingBottom: "calc(var(--w-aspect-h) / var(--w-aspect-w) * 100%)",
        ...aspectBase,
      })
    )
  )
  .case(
    "h",
    numberHandler<Record<keyof typeof spacingConfig, StyleObject> & Record<number, StyleObject>>(v =>
      css({
        "--w-aspect-h": v as string,
      })
    )
  )
  .use(
    genericHandler(v =>
      css({
        paddingBottom: +v * 100 + "%",
        ...aspectBase,
      })
    )
  )
  .init();
