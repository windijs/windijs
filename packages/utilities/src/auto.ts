import { gridAutoColumnsConfig, gridAutoRowsConfig } from "@windijs/config";
import { configHandler, createUtility } from "@windijs/core";

export default createUtility("auto")
  .case("rows", configHandler(gridAutoRowsConfig, "gridAutoRows"))
  .case("cols", configHandler(gridAutoColumnsConfig, "gridAutoColumns"))
  .init();
