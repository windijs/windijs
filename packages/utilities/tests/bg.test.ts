import { backgroundClipConfig, gradientDirectionConfig } from "@windijs/config";
import { backgroundGenericHandler, colorHandler, configHandler, createUtility } from "@windijs/core";
import { bg, colors, from, to, via } from "../src";
import { percent, prop } from "@windijs/helpers";

test("Background Color", () => {
  expect(bg.current.css).toMatchSnapshot();
  expect(bg.white.css).toMatchSnapshot();
  expect(bg.blue[700].css).toMatchSnapshot();
  expect(bg.red[500].css).toMatchSnapshot();
});

test("Background Color With Opacity", () => {
  expect(bg.black.opacity(50).css).toMatchSnapshot();
  expect(bg.amber[100].opacity(50).css).toMatchSnapshot();
});

test("Background Color With Gradient", () => {
  expect(bg.black.gradient.css).toMatchSnapshot();
  expect(bg.amber[100].gradient.css).toMatchSnapshot();
  expect(bg.black.opacity(90).gradient.css).toMatchSnapshot();
  expect(bg.amber[300].opacity(90).gradient.css).toMatchSnapshot();
});

test("Background Color With Different Opacity Name", () => {
  const bg = createUtility("bg")
    .use(colorHandler(colors, "backgroundColor", "--my-opacity"))
    .init();
  expect(bg.black.opacity(50).css).toMatchSnapshot();
  expect(bg.amber[100].opacity(50).css).toMatchSnapshot();
});

test("Background Color Without Opacity", () => {
  const bg = createUtility("bg")
    .use(colorHandler(colors, "backgroundColor"))
    .init();
  expect(bg.black.css).toMatchSnapshot();
  expect(bg.amber[100].css).toMatchSnapshot();
});

test("Background Color With Different Color Value Type", () => {
  const customColors = {
    inherit: "inherit",
    current: "currentColor",
    transparent: "transparent",
    black: "#000",
    dark: "#1c1c1e",
    rgb: "rgb(22, 13, 14)",
    rgba: "rgba(22, 13, 14, 0.5)",
    hsl: "hsl(360, 100%, 50%)",
    hsla: "hsla(360, 100%, 50%, 0.3)",
    hwb: "hwb(194 0% 0%)",
    hwba: "hwb(194 0% 0% / .5)",
  };
  const bg = createUtility("bg")
    .use(colorHandler(customColors, "backgroundColor", "--w-bg-opacity"))
    .init();

  expect(bg.inherit.css).toMatchSnapshot("inherit");
  expect(bg.current.css).toMatchSnapshot("current");
  expect(bg.transparent.css).toMatchSnapshot("transparent");
  expect(bg.black.css).toMatchSnapshot("black");
  expect(bg.black.opacity(50).css).toMatchSnapshot("black-opacity");
  expect(bg.dark.css).toMatchSnapshot("dark");
  expect(bg.dark.opacity(50).css).toMatchSnapshot("dark-opacity");
  expect(bg.rgb.css).toMatchSnapshot("rgb");
  expect(bg.rgb.opacity(50).css).toMatchSnapshot("rgb-opacity");
  expect(bg.rgba.css).toMatchSnapshot("rgba");
  expect(bg.rgba.opacity(50).css).toMatchSnapshot("rgba-opacity");
  expect(bg.hsl.css).toMatchSnapshot("hsl");
  expect(bg.hsl.opacity(50).css).toMatchSnapshot("hsl-opacity");
  expect(bg.hsla.css).toMatchSnapshot("hsla");
  expect(bg.hsla.opacity(50).css).toMatchSnapshot("hsla-opacity");
  expect(bg.hwb.css).toMatchSnapshot("hwb");
  expect(bg.hwb.opacity(50).css).toMatchSnapshot("hwb-opacity");
  expect(bg.hwba.css).toMatchSnapshot("hwba");
  expect(bg.hwba.opacity(50).css).toMatchSnapshot("hwba-opacity");
});

test("Background Color Without Opacity With Different Color Value Type", () => {
  const customColors = {
    inherit: "inherit",
    current: "currentColor",
    transparent: "transparent",
    black: "#000",
    dark: "#1c1c1e",
    rgb: "rgb(22, 13, 14)",
    rgba: "rgba(22, 13, 14, 0.5)",
    hsl: "hsl(360, 100%, 50%)",
    hsla: "hsla(360, 100%, 50%, 0.3)",
    hwb: "hwb(194 0% 0%)",
    hwba: "hwb(194 0% 0% / .5)",
  };
  const bg = createUtility("bg")
    .use(colorHandler(customColors, "backgroundColor"))
    .init();

  expect(bg.inherit.css).toMatchSnapshot("inherit");
  expect(bg.current.css).toMatchSnapshot("current");
  expect(bg.transparent.css).toMatchSnapshot("transparent");
  expect(bg.black.css).toMatchSnapshot("black");
  expect(bg.dark.css).toMatchSnapshot("dark");
  expect(bg.rgb.css).toMatchSnapshot("rgb");
  expect(bg.rgba.css).toMatchSnapshot("rgba");
  expect(bg.hsl.css).toMatchSnapshot("hsl");
  expect(bg.hsla.css).toMatchSnapshot("hsla");
  expect(bg.hwb.css).toMatchSnapshot("hwb");
  expect(bg.hwba.css).toMatchSnapshot("hwba");
});

test("Background Opacity", () => {
  expect(bg.blue[300].css).toMatchSnapshot();
  expect(bg.opacity[0].css).toMatchSnapshot();
  expect(bg.opacity[20].css).toMatchSnapshot();
});

test("Background Attachment", () => {
  expect(bg.fixed.css).toMatchSnapshot();
  expect(bg.local.css).toMatchSnapshot();
  expect(bg.scroll.css).toMatchSnapshot();
});

test("Background Clip", () => {
  expect(bg.clip.border.css).toMatchSnapshot();
  expect(bg.clip.padding.css).toMatchSnapshot();
  expect(bg.clip.content.css).toMatchSnapshot();
  expect(bg.clip.text.css).toMatchSnapshot();
});

test("Background Clip With Different Key", () => {
  const bg = createUtility("bg")
    .case("clipped", configHandler(backgroundClipConfig, ["backgroundClip", prop`-webkit-background-clip`]))
    .init();
  expect(bg.clipped.border.css).toMatchSnapshot();
  expect(bg.clipped.padding.css).toMatchSnapshot();
  expect(bg.clipped.content.css).toMatchSnapshot();
  expect(bg.clipped.text.css).toMatchSnapshot();
});

test("Background Position", () => {
  expect(bg.bottom.css).toMatchSnapshot();
  expect(bg.left.bottom.css).toMatchSnapshot();
});

test("Background Repeat", () => {
  expect(bg.repeat.css).toMatchSnapshot();
  expect(bg.repeat.x.css).toMatchSnapshot();
  expect(bg.repeat.y.css).toMatchSnapshot();
  expect(bg.repeat.round.css).toMatchSnapshot();
  expect(bg.repeat.space.css).toMatchSnapshot();
  expect(bg.noRepeat.css).toMatchSnapshot();
});

test("Background Size", () => {
  expect(bg.auto.css).toMatchSnapshot();
  expect(bg.cover.css).toMatchSnapshot();
  expect(bg.contain.css).toMatchSnapshot();
});

test("Background Origin", () => {
  expect(bg.origin.border.css).toMatchSnapshot();
  expect(bg.origin.padding.css).toMatchSnapshot();
  expect(bg.origin.content.css).toMatchSnapshot();
});

test("Background Blend Mode", () => {
  expect(bg.blend.luminosity.css).toMatchSnapshot();
  expect(bg.blend.softLight.css).toMatchSnapshot();
  expect(bg.blend.color.css).toMatchSnapshot();
  expect(bg.blend.color.burn.css).toMatchSnapshot();
  expect(bg.blend.color.dodge.css).toMatchSnapshot();
});

test("Background Generic", () => {
  const bg = createUtility("bg")
    .use(backgroundGenericHandler())
    .init();
  expect(bg["rgb(22, 22, 22)"].css).toMatchSnapshot();
  expect(bg.red.css).toMatchSnapshot();
  expect(bg.aliceBlue.css).toMatchSnapshot();
  expect(bg[0xffffff].css).toMatchSnapshot();
});

test("Background With Deep Nested Color", () => {
  const nestedColors = {
    white: "#fff",
    rose: {
      50: "#fff1f2",
      dark: {
        100: "#fff",
        night: {
          300: "#1c1c1e",
          blue: {
            600: "#222222",
          },
        },
      },
    },
  };

  const bg = createUtility("bg")
    .use(colorHandler(nestedColors, "backgroundColor", "--w-bg-opacity"))
    .init();

  expect(bg.white.css).toMatchSnapshot();
  expect(bg.rose[50].css).toMatchSnapshot();
  expect(bg.rose.dark[100].css).toMatchSnapshot();
  expect(bg.rose.dark.night[300].css).toMatchSnapshot();
  expect(bg.rose.dark.night.blue[600].css).toMatchSnapshot();

  expect(bg.white.meta.props).toMatchSnapshot();
  expect(bg.rose[50].meta.props).toMatchSnapshot();
  expect(bg.rose.dark[100].meta.props).toMatchSnapshot();
  expect(bg.rose.dark.night[300].meta.props).toMatchSnapshot();
  expect(bg.rose.dark.night.blue[600].meta.props).toMatchSnapshot();
});

// TODO: support radial gradient

test("Gradients", () => {
  expect(bg.none.css).toMatchSnapshot();
  expect(bg.gradient.to.t.css).toMatchSnapshot();
  expect(bg.gradient.to.tr.css).toMatchSnapshot();
  expect(bg.gradient.to.r.css).toMatchSnapshot();
  expect(bg.gradient.to.br.css).toMatchSnapshot();
  expect(bg.gradient.to.b.css).toMatchSnapshot();
  expect(bg.gradient.to.bl.css).toMatchSnapshot();
  expect(bg.gradient.to.l.css).toMatchSnapshot();
  expect(bg.gradient.to.tl.css).toMatchSnapshot();
  expect(bg.gradient.deg[15].css).toMatchSnapshot();
  expect(bg.gradient.deg[90].css).toMatchSnapshot();
  expect(bg.gradient.deg[120].css).toMatchSnapshot();
  expect(bg.gradient.aqua.css).toMatchSnapshot();
  expect(bg.gradient.sky.css).toMatchSnapshot();
});

test("Gradient Function", () => {
  const to = gradientDirectionConfig.to;
  expect(bg.gradient.to.tl.css).toMatchSnapshot();
  expect(bg.gradient.deg[15].css).toMatchSnapshot();
  expect(bg.gradient.aqua.css).toMatchSnapshot();
  expect(bg.gradient(to.br, colors.red[500], [colors.blue[500], percent[10]]).css).toMatchSnapshot();
  expect(bg.gradient(to.br, colors.amber[200], colors.green[200], colors.dark[100], colors.gray[200]).css).toMatchSnapshot();
});

test("Gradient From", () => {
  expect(from.yellow[500].css).toMatchSnapshot();
  expect(from.blue[500].opacity(10).css).toMatchSnapshot();
});

test("Gradient Via", () => {
  expect(via.yellow[500].css).toMatchSnapshot();
  expect(via.blue[500].opacity(10).css).toMatchSnapshot();
});

test("Gradient To", () => {
  expect(to.yellow[500].css).toMatchSnapshot();
  expect(to.yellow[500].opacity(20).css).toMatchSnapshot();
});
