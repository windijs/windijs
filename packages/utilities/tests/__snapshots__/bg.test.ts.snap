// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background Attachment 1`] = `
Object {
  "backgroundAttachment": "fixed",
}
`;

exports[`Background Attachment 2`] = `
Object {
  "backgroundAttachment": "local",
}
`;

exports[`Background Attachment 3`] = `
Object {
  "backgroundAttachment": "scroll",
}
`;

exports[`Background Blend Mode 1`] = `
Object {
  "backgroundBlendMode": "luminosity",
}
`;

exports[`Background Blend Mode 2`] = `
Object {
  "backgroundBlendMode": "soft-light",
}
`;

exports[`Background Blend Mode 3`] = `
Object {
  "backgroundBlendMode": "color",
}
`;

exports[`Background Blend Mode 4`] = `
Object {
  "backgroundBlendMode": "color-burn",
}
`;

exports[`Background Blend Mode 5`] = `
Object {
  "backgroundBlendMode": "color-dodge",
}
`;

exports[`Background Clip 1`] = `
Object {
  "-webkit-background-clip": "border-box",
  "backgroundClip": "border-box",
}
`;

exports[`Background Clip 2`] = `
Object {
  "-webkit-background-clip": "padding-box",
  "backgroundClip": "padding-box",
}
`;

exports[`Background Clip 3`] = `
Object {
  "-webkit-background-clip": "content-box",
  "backgroundClip": "content-box",
}
`;

exports[`Background Clip 4`] = `
Object {
  "-webkit-background-clip": "text",
  "backgroundClip": "text",
}
`;

exports[`Background Clip With Different Key 1`] = `
Object {
  "-webkit-background-clip": "border-box",
  "backgroundClip": "border-box",
}
`;

exports[`Background Clip With Different Key 2`] = `
Object {
  "-webkit-background-clip": "padding-box",
  "backgroundClip": "padding-box",
}
`;

exports[`Background Clip With Different Key 3`] = `
Object {
  "-webkit-background-clip": "content-box",
  "backgroundClip": "content-box",
}
`;

exports[`Background Clip With Different Key 4`] = `
Object {
  "-webkit-background-clip": "text",
  "backgroundClip": "text",
}
`;

exports[`Background Color 1`] = `
Object {
  "backgroundColor": "currentColor",
}
`;

exports[`Background Color 2`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(255, 255, 255, var(--w-bg-opacity))",
}
`;

exports[`Background Color 3`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(29, 78, 216, var(--w-bg-opacity))",
}
`;

exports[`Background Color 4`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(239, 68, 68, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: black 1`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(0, 0, 0, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: black-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(0, 0, 0, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: current 1`] = `
Object {
  "backgroundColor": "currentColor",
}
`;

exports[`Background Color With Different Color Value Type: dark 1`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(28, 28, 30, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: dark-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(28, 28, 30, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hsl 1`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "hsla(360, 100%, 50%, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hsl-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "hsla(360, 100%, 50%, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hsla 1`] = `
Object {
  "--w-bg-opacity": "0.3",
  "backgroundColor": "hsla(360, 100%, 50%, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hsla-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "hsla(360, 100%, 50%, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hwb 1`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "hwb(194 0% 0% / var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hwb-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "hwb(194 0% 0% / var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hwba 1`] = `
Object {
  "--w-bg-opacity": ".5",
  "backgroundColor": "hwb(194 0% 0% / var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hwba-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "hwb(194 0% 0% / var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: inherit 1`] = `
Object {
  "backgroundColor": "inherit",
}
`;

exports[`Background Color With Different Color Value Type: rgb 1`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(22, 13, 14, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: rgb-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(22, 13, 14, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: rgba 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(22, 13, 14, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: rgba-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(22, 13, 14, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: transparent 1`] = `
Object {
  "backgroundColor": "transparent",
}
`;

exports[`Background Color With Different Opacity Name 1`] = `
Object {
  "--my-opacity": "0.5",
  "backgroundColor": "rgba(0, 0, 0, var(--my-opacity))",
}
`;

exports[`Background Color With Different Opacity Name 2`] = `
Object {
  "--my-opacity": "0.5",
  "backgroundColor": "rgba(254, 243, 199, var(--my-opacity))",
}
`;

exports[`Background Color With Gradient 1`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(0, 0, 0, var(--w-bg-opacity))",
  "backgroundImage": "linear-gradient(180deg, rgba(255, 255, 255, 0.15), rgba(255, 255, 255, 0))",
}
`;

exports[`Background Color With Gradient 2`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(254, 243, 199, var(--w-bg-opacity))",
  "backgroundImage": "linear-gradient(180deg, rgba(255, 255, 255, 0.15), rgba(255, 255, 255, 0))",
}
`;

exports[`Background Color With Gradient 3`] = `
Object {
  "--w-bg-opacity": "0.9",
  "backgroundColor": "rgba(0, 0, 0, var(--w-bg-opacity))",
  "backgroundImage": "linear-gradient(180deg, rgba(255, 255, 255, 0.15), rgba(255, 255, 255, 0))",
}
`;

exports[`Background Color With Gradient 4`] = `
Object {
  "--w-bg-opacity": "0.9",
  "backgroundColor": "rgba(252, 211, 77, var(--w-bg-opacity))",
  "backgroundImage": "linear-gradient(180deg, rgba(255, 255, 255, 0.15), rgba(255, 255, 255, 0))",
}
`;

exports[`Background Color With Opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(0, 0, 0, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Opacity 2`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(254, 243, 199, var(--w-bg-opacity))",
}
`;

exports[`Background Color Without Opacity 1`] = `
Object {
  "backgroundColor": "#000",
}
`;

exports[`Background Color Without Opacity 2`] = `
Object {
  "backgroundColor": "#fef3c7",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: black 1`] = `
Object {
  "backgroundColor": "#000",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: current 1`] = `
Object {
  "backgroundColor": "currentColor",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: dark 1`] = `
Object {
  "backgroundColor": "#1c1c1e",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: hsl 1`] = `
Object {
  "backgroundColor": "hsl(360, 100%, 50%)",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: hsla 1`] = `
Object {
  "backgroundColor": "hsla(360, 100%, 50%, 0.3)",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: hwb 1`] = `
Object {
  "backgroundColor": "hwb(194 0% 0%)",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: hwba 1`] = `
Object {
  "backgroundColor": "hwb(194 0% 0% / .5)",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: inherit 1`] = `
Object {
  "backgroundColor": "inherit",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: rgb 1`] = `
Object {
  "backgroundColor": "rgb(22, 13, 14)",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: rgba 1`] = `
Object {
  "backgroundColor": "rgba(22, 13, 14, 0.5)",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: transparent 1`] = `
Object {
  "backgroundColor": "transparent",
}
`;

exports[`Background Generic 1`] = `
Object {
  "backgroundColor": "rgb(22, 22, 22)",
}
`;

exports[`Background Generic 2`] = `
Object {
  "backgroundColor": "red",
}
`;

exports[`Background Generic 3`] = `
Object {
  "backgroundColor": "aliceBlue",
}
`;

exports[`Background Generic 4`] = `
Object {
  "backgroundColor": "#ffffff",
}
`;

exports[`Background Opacity 1`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(147, 197, 253, var(--w-bg-opacity))",
}
`;

exports[`Background Opacity 2`] = `
Object {
  "--w-bg-opacity": "0",
}
`;

exports[`Background Opacity 3`] = `
Object {
  "--w-bg-opacity": "0.2",
}
`;

exports[`Background Origin 1`] = `
Object {
  "backgroundOrigin": "border-box",
}
`;

exports[`Background Origin 2`] = `
Object {
  "backgroundOrigin": "padding-box",
}
`;

exports[`Background Origin 3`] = `
Object {
  "backgroundOrigin": "content-box",
}
`;

exports[`Background Position 1`] = `
Object {
  "backgroundPosition": "bottom",
}
`;

exports[`Background Position 2`] = `
Object {
  "backgroundPosition": "left bottom",
}
`;

exports[`Background Repeat 1`] = `
Object {
  "backgroundRepeat": "repeat",
}
`;

exports[`Background Repeat 2`] = `
Object {
  "backgroundRepeat": "repeat-x",
}
`;

exports[`Background Repeat 3`] = `
Object {
  "backgroundRepeat": "repeat-y",
}
`;

exports[`Background Repeat 4`] = `
Object {
  "backgroundRepeat": "round",
}
`;

exports[`Background Repeat 5`] = `
Object {
  "backgroundRepeat": "space",
}
`;

exports[`Background Repeat 6`] = `
Object {
  "backgroundRepeat": "no-repeat",
}
`;

exports[`Background Size 1`] = `
Object {
  "backgroundSize": "auto",
}
`;

exports[`Background Size 2`] = `
Object {
  "backgroundSize": "cover",
}
`;

exports[`Background Size 3`] = `
Object {
  "backgroundSize": "contain",
}
`;

exports[`Background With Deep Nested Color 1`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(255, 255, 255, var(--w-bg-opacity))",
}
`;

exports[`Background With Deep Nested Color 2`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(255, 241, 242, var(--w-bg-opacity))",
}
`;

exports[`Background With Deep Nested Color 3`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(255, 255, 255, var(--w-bg-opacity))",
}
`;

exports[`Background With Deep Nested Color 4`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(28, 28, 30, var(--w-bg-opacity))",
}
`;

exports[`Background With Deep Nested Color 5`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(34, 34, 34, var(--w-bg-opacity))",
}
`;

exports[`Background With Deep Nested Color 6`] = `
Array [
  "white",
]
`;

exports[`Background With Deep Nested Color 7`] = `
Array [
  "rose",
  "50",
]
`;

exports[`Background With Deep Nested Color 8`] = `
Array [
  "rose",
  "dark",
  "100",
]
`;

exports[`Background With Deep Nested Color 9`] = `
Array [
  "rose",
  "dark",
  "night",
  "300",
]
`;

exports[`Background With Deep Nested Color 10`] = `
Array [
  "rose",
  "dark",
  "night",
  "blue",
  "600",
]
`;

exports[`Gradient From 1`] = `
Object {
  "--w-from-opacity": "1",
  "--w-gradient-from": "rgba(234, 179, 8, var(--w-from-opacity))",
  "--w-gradient-stops": "var(--w-gradient-from), var(--w-gradient-to, rgba(255, 255, 255, 0))",
}
`;

exports[`Gradient From 2`] = `
Object {
  "--w-from-opacity": "0.1",
  "--w-gradient-from": "rgba(59, 130, 246, var(--w-from-opacity))",
  "--w-gradient-stops": "var(--w-gradient-from), var(--w-gradient-to, rgba(255, 255, 255, 0))",
}
`;

exports[`Gradient Function 1`] = `
Object {
  "backgroundImage": Array [
    "-webkit-gradient(linear, right bottom, left top, from(var(--w-gradient-stops)))",
    "-o-linear-gradient(bottom right, var(--w-gradient-stops))",
    "linear-gradient(to top left, var(--w-gradient-stops))",
  ],
}
`;

exports[`Gradient Function 2`] = `
Object {
  "backgroundImage": Array [
    "-o-linear-gradient(75deg, var(--w-gradient-stops))",
    "linear-gradient(15deg, var(--w-gradient-stops))",
  ],
}
`;

exports[`Gradient Function 3`] = `
Object {
  "backgroundImage": "linear-gradient(135deg, #2096ff, #05ffa3)",
}
`;

exports[`Gradient Function 4`] = `
Object {
  "backgroundImage": Array [
    "-webkit-gradient(linear, left top, right bottom, from(#ef4444), color-stop(10%, #3b82f6))",
    "-o-linear-gradient(top left, #ef4444, #3b82f6 10%)",
    "linear-gradient(to bottom right, #ef4444, #3b82f6 10%)",
  ],
}
`;

exports[`Gradient Function 5`] = `
Object {
  "backgroundImage": Array [
    "-webkit-gradient(linear, left top, right bottom, from(#fde68a), color-stop(#bbf7d0), color-stop(#3c3c3c), to(#e5e7eb))",
    "-o-linear-gradient(top left, #fde68a, #bbf7d0, #3c3c3c, #e5e7eb)",
    "linear-gradient(to bottom right, #fde68a, #bbf7d0, #3c3c3c, #e5e7eb)",
  ],
}
`;

exports[`Gradient To 1`] = `
Object {
  "--w-gradient-to": "rgba(234, 179, 8, var(--w-to-opacity))",
  "--w-to-opacity": "1",
}
`;

exports[`Gradient To 2`] = `
Object {
  "--w-gradient-to": "rgba(234, 179, 8, var(--w-to-opacity))",
  "--w-to-opacity": "0.2",
}
`;

exports[`Gradient Via 1`] = `
Object {
  "--w-gradient-stops": "var(--w-gradient-from), rgba(234, 179, 8, var(--w-via-opacity)), var(--w-gradient-to, rgba(255, 255, 255, 0))",
}
`;

exports[`Gradient Via 2`] = `
Object {
  "--w-gradient-stops": "var(--w-gradient-from), rgba(59, 130, 246, var(--w-via-opacity)), var(--w-gradient-to, rgba(255, 255, 255, 0))",
}
`;

exports[`Gradients 1`] = `
Object {
  "backgroundImage": "none",
}
`;

exports[`Gradients 2`] = `
Object {
  "backgroundImage": Array [
    "-webkit-gradient(linear, left bottom, left top, from(var(--w-gradient-stops)))",
    "-o-linear-gradient(bottom, var(--w-gradient-stops))",
    "linear-gradient(to top, var(--w-gradient-stops))",
  ],
}
`;

exports[`Gradients 3`] = `
Object {
  "backgroundImage": Array [
    "-webkit-gradient(linear, left bottom, right top, from(var(--w-gradient-stops)))",
    "-o-linear-gradient(bottom left, var(--w-gradient-stops))",
    "linear-gradient(to top right, var(--w-gradient-stops))",
  ],
}
`;

exports[`Gradients 4`] = `
Object {
  "backgroundImage": Array [
    "-webkit-gradient(linear, left top, right top, from(var(--w-gradient-stops)))",
    "-o-linear-gradient(left, var(--w-gradient-stops))",
    "linear-gradient(to right, var(--w-gradient-stops))",
  ],
}
`;

exports[`Gradients 5`] = `
Object {
  "backgroundImage": Array [
    "-webkit-gradient(linear, left top, right bottom, from(var(--w-gradient-stops)))",
    "-o-linear-gradient(top left, var(--w-gradient-stops))",
    "linear-gradient(to bottom right, var(--w-gradient-stops))",
  ],
}
`;

exports[`Gradients 6`] = `
Object {
  "backgroundImage": Array [
    "-webkit-gradient(linear, left top, left bottom, from(var(--w-gradient-stops)))",
    "-o-linear-gradient(top, var(--w-gradient-stops))",
    "linear-gradient(to bottom, var(--w-gradient-stops))",
  ],
}
`;

exports[`Gradients 7`] = `
Object {
  "backgroundImage": Array [
    "-webkit-gradient(linear, right top, left bottom, from(var(--w-gradient-stops)))",
    "-o-linear-gradient(top right, var(--w-gradient-stops))",
    "linear-gradient(to bottom left, var(--w-gradient-stops))",
  ],
}
`;

exports[`Gradients 8`] = `
Object {
  "backgroundImage": Array [
    "-webkit-gradient(linear, right top, left top, from(var(--w-gradient-stops)))",
    "-o-linear-gradient(right, var(--w-gradient-stops))",
    "linear-gradient(to left, var(--w-gradient-stops))",
  ],
}
`;

exports[`Gradients 9`] = `
Object {
  "backgroundImage": Array [
    "-webkit-gradient(linear, right bottom, left top, from(var(--w-gradient-stops)))",
    "-o-linear-gradient(bottom right, var(--w-gradient-stops))",
    "linear-gradient(to top left, var(--w-gradient-stops))",
  ],
}
`;

exports[`Gradients 10`] = `
Object {
  "backgroundImage": Array [
    "-o-linear-gradient(75deg, var(--w-gradient-stops))",
    "linear-gradient(15deg, var(--w-gradient-stops))",
  ],
}
`;

exports[`Gradients 11`] = `
Object {
  "backgroundImage": Array [
    "-webkit-gradient(linear, left top, right top, from(var(--w-gradient-stops)))",
    "-o-linear-gradient(left, var(--w-gradient-stops))",
    "linear-gradient(90deg, var(--w-gradient-stops))",
  ],
}
`;

exports[`Gradients 12`] = `
Object {
  "backgroundImage": Array [
    "-o-linear-gradient(330deg, var(--w-gradient-stops))",
    "linear-gradient(120deg, var(--w-gradient-stops))",
  ],
}
`;

exports[`Gradients 13`] = `
Object {
  "backgroundImage": "linear-gradient(135deg, #2096ff, #05ffa3)",
}
`;

exports[`Gradients 14`] = `
Object {
  "backgroundImage": "linear-gradient(135deg, rgb(171, 220, 255) 10%, rgb(3, 150, 255) 100%)",
}
`;
